src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -762,72 +772,95 @@ public async Task Delete()
 
-                        await ServerStore.Cluster.WaitForIndexNotification(index, remaining);
-                        actualDeletionIndex = index;
-                    }
-                    catch (TimeoutException)
-                    {
-                        databaseIndex++;
-                    }
-                }
+                var lastRemoveIndex = await WaitForRemovingNodesFromDatabases(databasesToDelete, lastDeletionIndex, timeToWaitForConfirmation);
 
-                await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+                await using ( var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
                 {
                     context.Write(writer, new DynamicJsonValue
                     {
                         // we only send the successful index here, we might fail to delete the index
                         // because a node is down, and we don't want to cause the client to wait on an
                         // index that doesn't exists in the Raft log
-                        [nameof(DeleteDatabaseResult.RaftCommandIndex)] = actualDeletionIndex,
+                        [nameof(DeleteDatabaseResult.RaftCommandIndex)] = lastRemoveIndex,
                         [nameof(DeleteDatabaseResult.PendingDeletes)] = new DynamicJsonArray(pendingDeletes)
                     });
                 }
             }
         }
 
+        private async Task<long> WaitForRemovingNodesFromDatabases(
+            Dictionary<string, DeleteDatabaseDetails> databasesToDelete,
+            long lastDeletionIndex,
+            TimeSpan timeout)
+        {
+            if (databasesToDelete.Count == 0)
+                return -1;
+
+            await ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.GreaterOrEqual, lastDeletionIndex);
+
+            long lastRemoveIndex = 0;
+
+            var reqExecutors = new Dictionary<string, ClusterRequestExecutor>();
+            try
+            {
+                foreach (var (dbName, details) in databasesToDelete)
+                {
+                    var deleteCmdIndex = details.RaftIndex;
+
+                    foreach (var removedNodeTag in details.NodesToRemove) // nodeTags
+                    {
+                        var removeNodeFromDbCmdRaftIndex = 0L;
+                        var sw = Stopwatch.StartNew();
+                        while (TryGetLastRemoveNodeFromDatabaseCommand(ServerStore, dbName, removedNodeTag, deleteCmdIndex, out removeNodeFromDbCmdRaftIndex) == false)
+                        {
+                            if (sw.Elapsed > timeout)
+                            {
+                                throw new InvalidOperationException($"RemoveNodeFromDatabase command of node {removedNodeTag} doesn't exist in the log of node {ServerStore.NodeTag}.");
```

Comment:
this should be a timeout exception..

Reply:
No, because the timeout exception will cause the client request-executor to perform a failover, which I don't want

Reply:
So let's create a dedicated exception `DatabaseDeletionTimeoutException : TimeoutException` and handle it properly in new client.
Old clients will have the same behavior as before.
We should able to handle failover of this command anyway. 

@ppekrol thoughts?

Reply:
Done
I throw 'RavenTimeoutException'